crates/sdk/src/state.rs

```diff
@@ -2,13 +2,18 @@ use borsh::{BorshDeserialize, BorshSerialize};
 
 use crate::event::AppEvent;
 
-pub trait AppState: Default + BorshSerialize + BorshDeserialize //+ AppStateInit
-{
+pub trait AppState: Default + BorshSerialize + BorshDeserialize + AppStateInit {
     type Event<'a>: AppEvent + 'a;
 }
 
+pub trait Identity<This> {}
+
+impl<T> Identity<T> for T {}
```

Comment:
It would throw no errors

For example if you just write:
```
impl AppStateInit for KvStore {
    type Return = KvStore;
}
```

It throws no errors